### PR TITLE
ci(workflow): add auto-merge-pr workflow

### PR DESCRIPTION
**Pull request by hleonardtek**
To enhance automation, this PR introduces automatic merge of PR's with all approvals.

### CODE REVIEW

File: src/.github/workflows/auto-merge-pr.yaml
Comment (ccamel) [inline src/.github/workflows/auto-merge-pr.yaml:20]: By convention, we don't name jobs, so I'd remove this line.

File: src/.github/workflows/auto-merge-pr.yaml
Comment (ccamel) [inline src/.github/workflows/auto-merge-pr.yaml:22]:
```suggestion
      - name: Merge the PR (after approval)
```

File: src/.github/workflows/auto-merge-pr.yaml
Comment (ad2ien) [inline src/.github/workflows/auto-merge-pr.yaml:32]: let's delete the branch don't you think?

**Review by ad2ien — COMMENTED** [review]
It's going to be nice 😃

**Review by ccamel — APPROVED** [review]
👍

**Review by ad2ien — APPROVED** [review]
thanks!

**Review by ad2ien — COMMENTED** [review]
Actually, @amimart don't you think we should add it to auto-merge.yml instead?

**Review by amimart — CHANGES_REQUESTED** [review]
I'm not convinced by the approach, this is a tricky and very sensible subject that should be managed carefully,, and potentially with different approach depending on the project.

I think the first step before considering automatic merge is the categorization of the PR (e.g. dependency update, feature, fix, breaking change, etc...). Then, depending on the category we can set reviewers and assignees, and eventually automatically merge with different policies regarding the category.

**Comment by hleonardtek** [comment]
Issue closed because we need to clarify the issue and categorize PR's.
For some use cases we don't want to merge the code as soon as we get all the approvals, there are consequences. For example, a merge in the platform-cloud repo may involve argo-cd to deploy stuff on k8s.

**Pull request closed without merging**
